pkg/plugins/golang/v4/api.go

```diff
@@ -146,14 +146,6 @@ func (p *createAPISubcommand) InjectResource(res *resource.Resource) error {
 		}
 	}
 
-	// Ensure that if any external API flag is set, both must be provided.
-	if len(p.options.ExternalAPIPath) != 0 || len(p.options.ExternalAPIDomain) != 0 {
-		if len(p.options.ExternalAPIPath) == 0 || len(p.options.ExternalAPIDomain) == 0 {
-			return errors.New("Both '--external-api-path' and '--external-api-domain' must be " +
-				"specified together when referencing an external API.")
-		}
-	}
-
```

Comment:
Hi @fischor,

In my opinion, groups should not be empty when dealing with non-core types in scaffolds. The concepts should stay aligned with Kubernetes itself. We cannot have Kubebuilder diverging from core Kubernetes principles. However, in order to not block others scenarios we should not make mandatory users provide the Domain.

What are your thoughts?

Reply:
I am refering to the "group" (`--group`) as used in kubebuilder, not the Kubernetes apiGroup.
The apiGroup will not be empty if you do `--domain cert-manager.io --group ""`, it will be `cert-manager.io` then.

My thoughts are that is does not make sense to have --domain and --group for external APIs, since these are kubebuilder specific entities and not all external APIs use that scheme. Maybe just --api-group is better to specify the (full) apiGroup for external APIs.